### ISSUE TITLE
False positive for useless-else-on-loop

### ISSUE THREAD

**Issue by mitar** [PyCQA/pylint#2407]
Fro the following code:

```python
def parse(input_value):
    for suffix, value in SUFFIXES.items():
        if input_value.endswith(suffix):
            return value
    else:
        return int(input_value)
```

I get:

```
W0120: Else clause on loop without a break statement (useless-else-on-loop)
```

I would claim this is not useless because it makes it clearer when is the second `return` called. It might be a style preference question. I think this should not be triggered if there is `return` or `break` in the loop.

```
pylint 2.1.1
astroid 2.0.3
Python 3.6.3 (default, Oct  3 2017, 21:45:48) 
[GCC 7.2.0]
```

**Comment by PCManticore** [comment]
This is definitely a style preference that this check won't account for. The additional `else` is superfluous here and the code will be clearer without it as you don't have to consider any particular semantics when reading the code for when the `else` clause is reached.
